packages/venia-ui/lib/components/CartPage/ProductListing/quantity.js

```diff
@@ -0,0 +1,159 @@
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const [isIncrementDisabled, setIsIncrementDisabled] = useState(
+        value <= min ? true : false
+    );
+
+    const [isDecrementDisabled, setIsDecrementDisabled] = useState(
+        value >= max ? true : false
+    );
+
+    useEffect(() => {
+        // TODO: initializes with wrong state
+        setHasUnsavedValue(true);
+    }, [value]);
+
+    useEffect(() => {
+        if (value <= min) {
+            setIsDecrementDisabled(true);
+        } else {
+            setIsDecrementDisabled(false);
+        }
+
+        if (value >= max) {
+            setIsIncrementDisabled(true);
+        } else {
+            setIsIncrementDisabled(false);
+        }
+    }, [isDecrementDisabled, isIncrementDisabled, value]);
+
+    const decrement = () => {
+        let newValue = parseInt(value);
+
+        if (min === null || newValue > min) {
+            newValue--;
+            setValue(newValue);
+            setIsDecrementDisabled(false);
+        } else {
+            setIsDecrementDisabled(true);
+            // TODO: use Toast?
+            console.log('Minimum: ' + min);
+        }
+    };
+
+    const getUpdateButton = () => {
+        return hasUnsavedValue ? (
+            <button className={classes.updateButton} onClick={handleSaveValue}>
+                Update
+            </button>
+        ) : null;
+    };
+
+    const handleChange = event => {
+        const newValue = event.target.value;
+
+        if (newValue === '' || isNaN(newValue) === false) {
+            setValue(newValue);
+        } else {
+            setValue(min);
+
+            // TODO: use Toast?
+            console.log('Please enter a number.');
+        }
+
+        // TODO: decide how to handle if the user types value beyond min/max
+    };
+
+    const handleSaveValue = () => {
+        setHasUnsavedValue(false);
+        // TODO: talonize me
+    };
+
+    const increment = () => {
+        let newValue = parseInt(value);
+
+        if (max === null || newValue < max) {
+            newValue++;
+            setValue(newValue);
+            setIsIncrementDisabled(false);
+        } else {
+            setIsIncrementDisabled(true);
+            // TODO: use Toast?
+            console.log('Maximum: ' + max);
+        }
+    };
+
+    return (
+        <div className={classes.root}>
+            <div className={classes.wrap}>
+                <label className={classes.label} htmlFor={item.id}>
+                    {label}
+                </label>
+
+                <button
+                    aria-label={'Decrease ' + label}
+                    className={classes.button_decrement}
+                    disabled={isDecrementDisabled}
+                    onClick={decrement}
+                    type="button"
+                >
+                    <Icon className={classes.icon} src={MinusIcon} size={22} />
+                </button>
+
+                <input
```

Comment:
@sirugh Yes, fields should use `Form` from `informed` and `Field` from `venia-ui`, rather than storing the value in local state.

Ordinarily, I'd also expect our existing input components to be used, but in this case we're building a fairly unique one.